addons/web/static/src/views/list/list_renderer.scss

```diff
@@ -67,6 +71,10 @@
             }
         }
 
+        @include media-breakpoint-down(sm) {
+            width: auto;
```

Comment:
I'd add a `min-width: 100%` for the (very-)unlikely scenario that a table is shorten than the viewport.
 
<img width="379" height="242" alt="image" src="https://github.com/user-attachments/assets/646733b7-0c85-48b2-8bdb-adbb5801c571" />